src/main/java/com/hivemq/cli/utils/MqttUtils.java

```diff
@@ -65,4 +74,52 @@
                     ", host='" + host + '\'' +
                     '}';
         }
+
+    // See http://docs.oasis-open.org/mqtt/mqtt/v5.0/cs02/mqtt-v5.0-cs02.html#_Toc514345331
+    public static @NotNull String buildRandomClientID(final int length) {
+        if (length < 0) {
+            throw new NegativeArraySizeException("Length of random client id has to be positive");
```

Comment:
IllegalArgumentException would be better imho, you are not creating an array here